assets/js/theme/global/currency-selector.js

```diff
@@ -2,6 +2,17 @@ import swal from './sweet-alert';
 import utils from '@bigcommerce/stencil-utils';
 
 export default function (cartId) {
+    if (!cartId) return;
+
+    let countOfHandlers;
+    const events = $._data($('[data-cart-currency-switch-url]')[0], 'events');
+
+    $.each(events, (type, handlers) => {
+        countOfHandlers = handlers.length;
+    });
+
+    if (countOfHandlers >= 1) return;
```

Comment:
Even though this method of using internal JQuery attribute to get handlers is suggested in many places, I wonder why can't we just do something like this?
```
let currencySelectorCalled = false;

export default function (cartId) {
    if (!cartId) {
        return;
    }

    if (!currencySelectorCalled) {
        currencySelectorCalled = true;
    } else {
        return;
    }
```

Reply:
You're right, thanks.